cms/taxonomy/management/commands/sync_topics.py

```diff
@@ -0,0 +1,211 @@
+
+        # Extract just the fields we need
+        subtopics = [
+            {
+                k: v
+                for k, v in raw_topic.items()
+                if k
+                in {
+                    "title",
+                    "id",
+                    "description",
+                }
+            }
+            for raw_topic in raw_topics
+        ]
+
+        for topic in subtopics:
+            if parent_id:
+                topic["parent_id"] = parent_id
+
+        # Extend the topics list, to build a flat list of topics and subtopics which contain their own parent IDs
+        topics.extend(subtopics)
+
+        # Add any subtopics URLs from the topics we just fetched to the stack
+        request_stack.extend(_extract_subtopic_links(raw_topics))
+
+    return topics
+
+
+def _request_topics(url: str) -> list[dict[str, Any]]:
+    """Fetch topics from the API and return the items from the response."""
+    topics_response = requests.get(url, timeout=30)
+    topics_response.raise_for_status()
+    raw_topics: list[dict[str, Any]] = topics_response.json().get("items", [])
+    return raw_topics
+
+
+def _extract_subtopic_links(raw_topics: list[dict[str, Any]]) -> list[tuple[str, str | None]]:
+    """Return a list of tuples of any subtopic links and their parent topic IDs found in raw_topics."""
+    return [
+        (subtopic_link, raw_topic["id"])
+        for raw_topic in raw_topics
+        # NOTE: This assumes we can simply call the links in the response as they are provided,
+        # perhaps switch to building the URL ourselves
+        if (
+            (subtopic_link := raw_topic.get("links", {}).get("subtopics", {}).get("href"))
+            and raw_topic.get("subtopics_ids")
+        )
+    ]
+
+
+def _sync_with_fetched_topics(fetched_topics: list[dict[str, str]]) -> None:
+    """For each fetched topic, decide if it needs to be created, updated, or left as is."""
+    _check_for_duplicate_topics(fetched_topics)
+
+    updated_count = 0
+    created_count = 0
+    for fetched_topic in fetched_topics:
+        if existing_topic := _get_topic(fetched_topic["id"]):
+            if not _topic_matches(existing_topic, fetched_topic):
+                _update_topic(existing_topic, fetched_topic)
+                updated_count += 1
+        else:
+            _create_topic(fetched_topic)
+            created_count += 1
+
+    logger.info("Saved %d new topic(s)", created_count)
+    logger.info("Updated %d existing topic(s)", updated_count)
+
+
+def _check_for_duplicate_topics(fetched_topics: list[dict[str, str]]) -> None:
+    topic_ids = set()
+    for topic in fetched_topics:
+        if topic["id"] in topic_ids:
+            raise RuntimeError(f"Received duplicate topic ID in API responses, topic IDs must be unique: {topic['id']}")
+        topic_ids.add(topic["id"])
+
+
+def _get_topic(topic_id: str) -> Topic | None:
+    """Fetches a Topic record by its primary key (id), or returns None if none is found."""
```

Comment:
```suggestion
    """Fetches a Topic record by its primary key (id), or returns None if not found."""
```

Reply:
Addressed in https://github.com/ONSdigital/dis-wagtail/pull/104/commits/89fb9c9c8c63153bba489e9abedc3084ec285310